### PR TITLE
Add version checker and refactor get contributors

### PR DESCRIPTION
**Pull request by RickvdStaaij**
### What

Add version checker and refactor get contributors

### To do

- [x] Reload the browser when a new server version is detected
    - Push the version in de `index.html`
    - Add endpoint to get the latest version
- [x] Add documentation on how to update
- [x] ~~Save the version in a separate file in `dist/`, so the docker container doesn't have to keep extra shit~~ not doing this, just keep the `package.json` for now.

### CODE REVIEW

File: server/domain/cimonitor/Contributors.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        });
+
+        return Promise.all(fetchUserDetailPromises).then(() => detailedContributors);
+    }
+
+    sortContributorsByCommits(contributors) {
+        return contributors.sort((contributorA, contributorB) => contributorB.commits - contributorA.commits);
+    }
+
+    removeT888(contributors) {
```
Comment (eXistenZNL) [inline server/domain/cimonitor/Contributors.js]: Smart :+1:

File: server/domain/cimonitor/VersionChecker.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    getLatestVersion() {
+        return GitHubApi.call('repos/CIMonitor/CIMonitor/releases/latest').then(latestVersionInfo => {
+            return latestVersionInfo.tag_name;
+        });
+    }
+
+    getCurrentVersion() {
+        const root = path.resolve(`${__dirname}/../../../`);
```
Comment (eXistenZNL) [inline server/domain/cimonitor/VersionChecker.js]: Hmm, not a big fan of this due to security implications... not sure how to deal with this in NodeJS either.

Comment (RickvdStaaij) [reply server/domain/cimonitor/VersionChecker.js]: What is the security implication you see?

**Comment by RickvdStaaij** [comment]
Fixed a lot of stuff, however I'm not sure yet if I actually want to ship this version checking stuff. need to sleep on it.

**Comment by RickvdStaaij** [comment]
I've slept on it, and I think this does add value. The server side is set up neatly, yet the client side could be cleaner.

@eXistenZNL what do you think about the current state?

File: server/domain/cimonitor/VersionChecker.js (hunk 2 above)
Comment (eXistenZNL) [reply server/domain/cimonitor/VersionChecker.js]: If somehow this file can be loaded from another directory, you can climb outside of your root, include other stuff, and possibly gain access to your server. Don't have a working example for you now but I feel going ../../../../ is never a really nice solution.

File: client/components/Dashboard/VersionChecker.js [hunk 3]
```diff
@@ -9,31 +9,37 @@ class VersionChecker {
...
+            this.refreshDashboard(result.version);
+            return;
         }
+
+        console.log(`Dashboard version ${this.version} up-to-date with server version ${result.version}.`);
     }
 
-    refreshDashboard() {
-        console.log('Refreshing in 10 seconds to be up-to-date');
+    refreshDashboard(serverVersion) {
```
Comment (eXistenZNL) [inline client/components/Dashboard/VersionChecker.js]: I would call this showUpdateNotificationOnDashboard :)

Comment (RickvdStaaij) [reply client/components/Dashboard/VersionChecker.js]: Here it actually refreshes the dashboard as it's the client side code :innocent:

File: server/domain/cimonitor/VersionChecker.js (hunk 2 above)
Comment (RickvdStaaij) [reply server/domain/cimonitor/VersionChecker.js]: It's a server-side application, you can't access the backend files in any way. Discussed in real life & agreed.

**Pull request merged**